plutus-ledger-api/src/PlutusLedgerApi/V1/Value.hs

```diff
@@ -53,36 +53,47 @@ import Prelude qualified as Haskell
 
 import Control.DeepSeq (NFData)
 import Data.ByteString qualified as BS
-import Data.Data
+import Data.Data (Data)
 import Data.String (IsString (fromString))
 import Data.Text (Text)
 import Data.Text qualified as Text
 import Data.Text.Encoding qualified as E
 import GHC.Generics (Generic)
 import PlutusLedgerApi.V1.Bytes (LedgerBytes (LedgerBytes), encodeByteString)
-import PlutusLedgerApi.V1.Scripts
-import PlutusTx qualified as PlutusTx
+import PlutusLedgerApi.V1.Scripts (MintingPolicyHash (..))
+import PlutusTx qualified
 import PlutusTx.AssocMap qualified as Map
 import PlutusTx.Lift (makeLift)
 import PlutusTx.Ord qualified as Ord
 import PlutusTx.Prelude as PlutusTx hiding (sort)
-import PlutusTx.These
-import Prettyprinter
-import Prettyprinter.Extras
+import PlutusTx.These (These (..))
+import Prettyprinter (Pretty, (<>))
+import Prettyprinter.Extras (PrettyShow (PrettyShow))
 
+{- | ByteString representing the currency, hashed with /BLAKE2b-224/.
+It is empty for `Ada`, 28 bytes for `PolicyId`.
+
+This is a simple type without any validation, __use with caution__.
+You may want to add checks for its invariants. See the
+ [Shelly ledger specification](https://hydra.iohk.io/build/16861845/download/1/ledger-spec.pdf).
+-}
 newtype CurrencySymbol = CurrencySymbol { unCurrencySymbol :: PlutusTx.BuiltinByteString }
-    deriving (IsString, Haskell.Show, Pretty) via LedgerBytes
+    deriving
+        (IsString        -- ^ from hex encoding
+        , Haskell.Show   -- ^ using hex encoding
+        , Pretty         -- ^ using hex encoding
+        ) via LedgerBytes
     deriving stock (Generic, Data)
     deriving newtype (Haskell.Eq, Haskell.Ord, Eq, Ord, PlutusTx.ToData, PlutusTx.FromData, PlutusTx.UnsafeFromData)
     deriving anyclass (NFData)
 
 {-# INLINABLE mpsSymbol #-}
--- | The currency symbol of a monetary policy hash
+-- | The `CurrencySymbol` of a monetary policy hash
```

Comment:
Oh yes, answering my own question policyid seems to be the same. Perhaps rename every PolicyId occurence to MintingPolicyHash